lxd/api_internal_recover.go

```diff
@@ -166,6 +168,67 @@ func appendUnknownVolumeConfig(originalPoolName string, projectName string, volu
 	return nil
 }
 
+// identifyCustomVolumePool checks if the respective pool of a discovered custom volume is already known or missing.
+// This can be the case if a custom volume was discovered through an instance's backup config on another pool.
+// It may return a potential dependency error as string in case the pool doesn't yet exist.
```

Comment:
this sounds funny, why are we returning special errors as strings rather than `error`s?